vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp

```diff
@@ -290,10 +290,13 @@ static YGValue YGValueOrDefault(const winrt::Microsoft::ReactNative::JSValue &va
       folly::dynamic pct(str);
       return YGValue{static_cast<float>(pct.asDouble()), YGUnitPercent};
     }
+    if (str.length() > 2 && (str.substr((str.length() - 2), 2) == "px" || str.substr((str.length() - 2), 2) == "pt")) {
+      shadowNode.RedBox("Value '" + static_cast<std::string>(value.AsString()) + "' for " + key + " is invalid. Cannot be converted to YGValue. '" + str.substr((str.length() - 2), 2) + "' unit not needed. Simply use integer value.");
```

Comment:
I don't think you want to `static_cast` to a value. Did you run into issues using the raw result of `value.AsString()`?

Reply:
I didn't give just value.AsString() a try; I believe I just copied the format from Line 291 for the float cast. Just tried the change now and it works. 👍

Reply:
For more context, `static_cast` has a couple of uses. One is to treat a pointer to one type as a pointer to another (e.g. casting from superclass to a subclass). Another is to change the representation of a primitive. E.g. converting between `double` and `float`.

Reply:
Ah yes makes sense. Thanks for the extra info!